### ISSUE TITLE
`context.origin` should contain the full sender's Bitcoin address

### ISSUE THREAD

**Issue by fadeev** [zeta-chain/node#1366]
Right now when a user calls an omnichain contract from Bitcoin, [`context.origin`](https://github.com/zeta-chain/protocol-contracts/blob/accb3cba156943beedc8a09a5a5c36e4bbcfffff/contracts/zevm/interfaces/zContract.sol#L12) only contains the first 20 bytes of the hex representation of the bech32 Bitcoin addres.

So, if the address is:

```
tb1q2dr85d57450xwde6560qyhj7zvzw9895hq25tx
```

The hex representation of it is:

```
0x74623171326472383564353734353078776465363536307179686a377a767a7739383935687132357478
```

`context.origin` is:

```
0x7462317132647238356435373435307877646536
```

Having the full hex address in `context.origin` will greatly improve the developer experience of creating omnichain contracts.

**Comment by fadeev** [comment]
Here's an example using the Staking contract:

https://zetachain-athens-3.blockscout.com/address/0xECf5f5009D0bCD9615d8aC2893BfaD9250EfD4da

I've just deployed a contract with Bitcoin as a counterparty chain and submitted one CCTX: to set the withdrawal Bitcoin address.

```
npx hardhat send-btc --amount 0.001 --memo ECf5f5009D0bCD9615d8aC2893BfaD9250EfD4da0474623171326472383564353734353078776465363536307179686a377a767a7739383935687132357478 --recipient tb1qy9pqmk2pd9sv63g27jt8r657wy0d9ueeh0nqur
```

Contents of memo:

* `ECf5f5009D0bCD9615d8aC2893BfaD9250EfD4da` omnichain contract address
* `04` integer indicating which action to perform ("set withdrawal address")
* `74623171326472383564353734353078776465363536307179686a377a767a7739383935687132357478` hex-encoded address of the sender's Bitcoin address: `tb1q2dr85d57450xwde6560qyhj7zvzw9895hq25tx`

The logic that gets executed:

https://github.com/zeta-chain/example-contracts/blob/859ce9950f30b9373c0ce7434080e6b54613a331/omnichain/staking/contracts/Staking.sol#L71

https://github.com/zeta-chain/example-contracts/blob/859ce9950f30b9373c0ce7434080e6b54613a331/omnichain/staking/contracts/Staking.sol#L130C14-L142

Basically, we're setting a mapping:

```
address staker = BytesHelperLib.bytesToAddress(context.origin, 0);
setWithdraw(staker, message, context.origin)
withdrawAddress = bytesToBech32Bytes(message, 1);
withdraw[staker] = withdrawAddress;
```

We're using `context.origin` as a key and the a value from the message as the actual address.

Here's the result of this tx:

<img width="1461" alt="Screenshot 2024-01-08 at 22 55 06" src="https://github.com/zeta-chain/node/assets/332151/0223908d-429b-440c-bbfc-1b7189724628">

If you query by what was passed as `context.origin` (`0x7462317132647238356435373435307877646536`) you will get the value passed in the message.

cc @brewmaster012

**Comment by fadeev** [comment]
Ok, I was wrong. Looks like `context.origin` indeed contains the whole address, the decoding function in the contract was truncating it.
